src/components/shared/ManageComponent/hooks/useComponentCanvasTasks.ts

```diff
@@ -0,0 +1,24 @@
+import { useMemo } from "react";
+
+import { useComponentSpec } from "@/providers/ComponentSpecProvider";
+import { isGraphImplementation } from "@/utils/componentSpec";
+
+/**
+ * Returns the tasks on the canvas that use the component with the given digest.
+ *
+ * @param digest - The digest of the component to find tasks for.
+ * @returns The tasks on the canvas that use the component with the given digest.
+ */
+export function useComponentCanvasTasks(digest: string) {
+  const { componentSpec } = useComponentSpec();
+
+  return useMemo(
+    () =>
+      isGraphImplementation(componentSpec.implementation)
+        ? Object.values(componentSpec.implementation?.graph?.tasks).filter(
+            (task) => task.componentRef.digest === digest,
+          )
+        : [],
+    [componentSpec.implementation, digest],
+  );
```

Comment:
I think this should work

```suggestion
  const { graphSpec } = useComponentSpec();

  return useMemo(
    () =>Object.values(graphSpec.tasks).filter(
            (task) => task.componentRef.digest === digest,
          ),
    [graphSpec, digest],
  );
```